freva-rest/src/freva_rest/databrowser_api/endpoints.py

```diff
@@ -304,8 +309,17 @@ async def extended_search(
         translate=translate,
         **SolrSchema.process_parameters(request),
     )
+    if (
+        zarr_stream
+        and current_user is None
+        and "zarr-stream" in server_config.services
```

Comment:
Have you thought about the case where `zarr-stream` is requested but it's not supported by the server, meaning that it's not in the `services` flag. This flag is for instances that don't want to go through the pain of setting up an extra redis and streaming worker. I assume that FUB woulnd't be interested in this feature so I don't want to make them deploy stuff they don't need.

I would do it like this:

```python
if zarr-stream not in server_config:
   zarr_stream = False
   
 if current_user is None and zarr_stream:
   raise HTTPException(
                        status_code=401,
                        detail="Zarr-streaming needs authentication",
                    )
```